src/iceberg/iceberg_parser/iceberg_parser.py

```diff
@@ -0,0 +1,209 @@
+"""
+Author: Ioannis Paraskevakos
+License: MIT
+Copyright: 2018-2019
+"""
+
+from __future__ import print_function
+import argparse
+import sys
+
+
+class IcebergParser(object):
+    """
+    This class is the argument parser of the ICEBERG software tool.
+    """
+
+    def __init__(self):
+        """
+        The constructor
+        """
+        parser = argparse.ArgumentParser(description='ICEBERG command description:',
+                                         epilog='''Enjoy our tool!''')
+        required_args = parser.add_argument_group()
+        required_args.title = 'Required Arguments'
+        required_args.add_argument('--resource', '-r', help='Where the execution will happen',
+                                   type=str)
+        required_args.add_argument('--queue', '-q', help='The queue of the resource',
+                                   type=str)
+        required_args.add_argument('--cpus', '-c', help='How many CPUs will be required',
+                                   type=int)
+        required_args.add_argument('--gpus', '-g', help='How many GPUs will be required',
+                                   type=int)
+        required_args.add_argument('--input_path', '-ip', help='Where the input images are',
+                                   type=str)
+        required_args.add_argument('--output_path', '-op', help='Where the results should be saved',
+                                   type=str)
+        required_args.add_argument('--walltime', '-w', help='The estimated execution time',
+                                   type=int)
+        required_args.add_argument('--analysis', '-a', help='The type of analysis to be executed',
+                                   type=str)
+
+        if len(sys.argv) < 9:
+            parser.print_help()
+            argv = sys.argv
+            if '--help' in argv:
+                argv.remove('--help')
+            elif '-h' in argv:
+                argv.remove('-h')
+
+            if len(argv) > 1:
+                args = parser.parse_args(argv[1:])
+                if args.analysis:
+                    usecase = args.analysis
+                    if usecase and usecase == '4DGeolocation':
+                        usecase = 'four_d_geolocation'
+                    if not hasattr(self, usecase):
+                        print('Unrecognized Analysis Type')
+                    else:
+                        getattr(self, usecase)(show_help=True)
+            exit(1)
+        else:
+            args = parser.parse_args(sys.argv[1:9])
+
+        if args.analysis == '4DGeolocation':
+            usecase = 'four_d_geolocation'
+        else:
+            usecase = args.analysis
+
+        if not hasattr(self, usecase):
```

Comment:
trying to test the iceberg args, doing the following : 
```iceberg --resource xsede.bridges --queue RM  --cpus 28 --gpus 2 --input_path /home/aymen/ --output_path /home/aymen/ --walltime 90 --analysis seals ```

or 

``` iceberg --resource xsede.bridges --queue RM  --cpus 28 --gpus 2 --input_path /home/aymen/ --output_path /home/aymen/ --walltime 90 --analysis penguins```

or 

``` iceberg --resource xsede.bridges --queue RM  --cpus 28 --gpus 2 --input_path /home/aymen/ --output_path /home/aymen/ --walltime 90 --analysis 4DGeolocation```

etc. 

getting the following : 
`File "/home/aymen/miniconda3/envs/conda_env1/bin/iceberg", line 12, in <module>
    IcebergParser()
  File "/home/aymen/miniconda3/envs/conda_env1/lib/python2.7/site-packages/iceberg/iceberg_parser/iceberg_parser.py", line 69, in __init__
    if not hasattr(self, usecase):
TypeError: hasattr(): attribute name must be string`

Reply:
This is interesting. You used `--queue RM` instead of `--queue=RM` that I prefer to use. That immediately increases the number of input arguments and as a result they do not get filtered correctly. This is something that will pop up if a user used the small case parameter flags.